### ISSUE TITLE
Can't set value for dynamic widgets inside components in screenshot mode

### ISSUE THREAD

**Issue by giladresisi** [FlutterFlow/flutterflow-issues#3390]
### Can we access your project?

- [X] I give permission for members of the FlutterFlow team to access and test my project for the sole purpose of investigating this issue.

### Current Behavior

I'm trying to set the value for all widgets in my app in screenshot mode, but as of now the popup which allows setting the value opens only for widgets residing directly within a scaffold (page), for items residing within components the popup doesn't open and I don't know how to set their value in screenshot mode.
Maybe I'm missing something but I didn't see anything related to this in the docs or in the relevant youtube video.

### Expected Behavior

I'd expect that even for widgets residing within components the popup for setting their value in screenshot mode will open to allow me to set their value.

### Steps to Reproduce

1. Add a container to a page
2. Add a text widget into this container
3. Convert the container to a component
4. Click "Preview App", click "Screenshot Mode" and click the text widget to try to set its value for screenshots - no popup opens

### Reproducible from Blank

- [X] The steps to reproduce above start from a blank project.

### Bug Report Code (Required)

IT8glPHqy5NOm99c1pD2LO9V9WIvKUA2aIIvjuxQEAgafJSvBI13f/T4dkxuQuqBam1cJWaGmm0zwPOKheDpCsE4FByJbbk41rp2ZT3wLHGXaJjWC5CKVUZSGf5VGHLFzpuZhSVDBOhZSCFn7E+UN+SiSgvDNYCwImozL8n/KrrOj3SkGR3eGn0Jg1ZSfCvr

### Visual documentation

![Screenshot 2024-07-13 095645](https://github.com/user-attachments/assets/f7f40e15-737e-4103-8d4c-f3d7deb3efc3)
![Screenshot 2024-07-13 101202](https://github.com/user-attachments/assets/6d941d0d-9e72-43df-b041-185c36c20cc7)

### Environment

```markdown
- FlutterFlow version: v4.1.72+
- Platform: Web
- Browser name and version: Chrome 126.0.6478.127
- Operating system and version affected: Windows 11 Pro
```

### Additional Information

I'm about to start publishing my app beyond my initial small test group and I can't publish it when the screenshots still show those ugly "If/Then/Else..." (see one of the attached photos I uploaded). I really need this solved ASAP so I can set the value for all relevant widgets of my app including those which reside within components.

**Comment by paulperez-dev** [comment]
Hi @giladresisi! I was unable to reproduce this issue. You shared a bug report code from a project where the issue is not present. Please, reproduce the error in your project and generate the bug report code again.

**Comment by giladresisi** [comment]
Hi, here's a bug report code from the project I'm working on:
IT4oz8r13ItOxLhc+KrAdfpVnSQ7Qkc3argo0MpBazwbfe+uB4kqevfOTERoXderS1Vme2CZkWoW0+DRkd/qVvA3IQyVRYhFys5+bg2VUj+lM5CkFJWgYEdBHcFgGhHF57e3kR5NAM52WUJmw2GEF9/KbBfaNZj1OVUsDq/LZO4=

**Comment by paulperez-dev** [comment]
Hi @giladresisi! Can you please confirm if you can change the value of the widget shown in the following image:

<img src="https://uploads.linear.app/3241089d-736a-4422-900b-051e72accdbe/1b2351d9-6355-41c7-b508-ef670c4436f0/e80e122c-7ac3-4a3a-86d9-db5ba4ac89b0?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiLzMyNDEwODlkLTczNmEtNDQyMi05MDBiLTA1MWU3MmFjY2RiZS8xYjIzNTFkOS02MzU1LTQxYzctYjUwOC1lZjY3MGM0NDM2ZjAvZTgwZTEyMmMtN2FjMy00YTNhLTg2ZDktZGI1YmE0YWM4OWIwIiwiaWF0IjoxNzIxMTg2ODQ5LCJleHAiOjMzMjkxNzQ2ODQ5fQ.Ya6-Jw0kDj2yYPOveG6-UyDGK4D2t20llR-9mLIWvAc " alt="image.png" width="610" height="724" />

**Comment by giladresisi** [comment]
For this one - yes, but for everything else in this page which relies on a
variable - no.

‫בתאריך יום ד׳, 17 ביולי 2024 ב-10:27 מאת ‪Paul Pérez‬‏ <‪
***@***.***‬‏>:‬

**Comment by paulperez-dev** [comment]
Hi @giladresisi! The bug is with the `RichText` widget, not the `Text` widget. I will send this to the ENG team so they can work on this. 

As a workaround, you can use the `UI Builder Display Value` to show the desired text on the UI (it also applies to the `Screenshot Mode`):

<img src="https://uploads.linear.app/3241089d-736a-4422-900b-051e72accdbe/d100e35e-4d5d-499d-b80c-14ab8ce13ca5/7eec641c-821b-44e8-85ea-7ebd3db2b8c0?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiLzMyNDEwODlkLTczNmEtNDQyMi05MDBiLTA1MWU3MmFjY2RiZS9kMTAwZTM1ZS00ZDVkLTQ5OWQtYjgwYy0xNGFiOGNlMTNjYTUvN2VlYzY0MWMtODIxYi00NGU4LTg1ZWEtN2ViZDNkYjJiOGMwIiwiaWF0IjoxNzIxNTY2MzE4LCJleHAiOjMzMjkyMTI2MzE4fQ.Teasxq9_GI1qxeIxq2DGseWJkwC4VzcPY6pd5LgNveA " alt="image.png" width="291" height="641" />

**Comment by giladresisi** [comment]
This doesn't work either for the RichText widget, I still see the
[variable_name] text instead of the value I put in the "UI Builder
Display Value".
Please make sure this is fixed as soon as possible, I'm going to start
marketing my app very soon and I need decent screenshots for this.
Thanks,
Gilad

‫בתאריך יום א׳, 21 ביולי 2024 ב-19:52 מאת ‪Paul Pérez‬‏ <‪
***@***.***‬‏>:‬

**Comment by giladresisi** [comment]
Hi, any news here? I can't upload screenshots containing variable names to
the stores

‫בתאריך יום א׳, 21 ביולי 2024 ב-21:03 מאת ‪Gilad Resisi‬‏ <‪
***@***.***‬‏>:‬

**Comment by paulperez-dev** [comment]
Thanks for the confirmation. The issue is the RichText widget, not the Text widget. I will send this to the ENG team so they can work on this. Thanks!

**Comment by giladresisi** [comment]
Of course, you guys said this long ago, why is it only sent now to the eng
team?
Please accelerate this, I need to create new screenshots ASAP.
Thanks,
Gilad

בתאריך יום ה׳, 15 באוג׳ 2024, 8:24, מאת Paul Pérez ‏<
***@***.***>:

**Comment by paulperez-dev** [comment]
It was sent 3 weeks ago but I noticed I did not let you know, so I wrote the previous message.

**Comment by giladresisi** [comment]
Hi,
This issue is really becoming urgent!
It's been a month now already since this was sent to the eng team, anything
new?
I'm being rejected by Apple when submitting new versions because my
screenshots have those variable names in them (due to this bug).
I really need to publish a new version in the next few days, please have it
solved ASAP, until then I'm pretty much stuck...
Thanks,
Gilad

בתאריך יום ה׳, 15 באוג׳ 2024, 10:56, מאת Paul Pérez ‏<
***@***.***>:

**Comment by giladresisi** [comment]
OK now I really really need this issue solved ASAP.
My marketing partners start their campaign for the app in 5 days from now
and it always takes a few days for Apple and Google to approve and publish
changes in the app stores.
I really need you guys to solve this, I already solved all other issues in
my app and Apple won't approve new versions until I have the screenshots
isdue solved.
Please help here ASAP!
Thanks,
Gilad

בתאריך יום ה׳, 22 באוג׳ 2024, 8:37, מאת Gilad Resisi ***@***.***

**Comment by paulperez-dev** [comment]
Hi @giladresisi! There is no update for the fix yet.

In the meantime, you can get your captures using one of the following alternatives/workarounds:

1. Run the app on a real device and get the capture from there.
2. Duplicate the screen and change all the `RichText` widgets for `Text` widgets (as they work as expected) and take the captures.

**Comment by adachoinw** [comment]
Hi
I can't change the screenshot value for richtext or text. The screenshot value popup doesn't come up anymore.
It was working last week. I tried different browsers and updated the flutterflow app, and they all have the same issue.
Are there any other workarounds other than running the app on a real device?
What is the ETA for the fix?
![2024-11-25 20_51_41-Preview - Ask Ada](https://github.com/user-attachments/assets/9828b1b3-86d7-4728-831e-3e62c25739ed)
